modular_skyrat/modules/tableflip/code/flipped_table.dm

```diff
@@ -64,25 +68,31 @@
 
 /obj/structure/table/CtrlShiftClick(mob/living/user)
 	. = ..()
-	if(!istype(user) || !user.can_interact_with(src) || isobserver(user) || iscorticalborer(user)) //skyrat edit: no borer flipping
+	if(!istype(user) || !user.can_interact_with(src) || isobserver(user) || iscorticalborer(user))
 		return
-	if(can_flip)
-		user.visible_message(span_danger("[user] starts flipping [src]!"), span_notice("You start flipping over the [src]!"))
-		if(do_after(user, max_integrity/4))
-			var/obj/structure/flippedtable/T = new flipped_table_type(src.loc)
-			T.name = "flipped [src.name]"
-			T.desc = "[src.desc] It is flipped!"
-			T.icon_state = src.base_icon_state
-			var/new_dir = get_dir(user, T)
-			T.dir = new_dir
-			if(new_dir == NORTH)
-				T.layer = BELOW_MOB_LAYER
-			T.max_integrity = src.max_integrity
-			T.update_integrity(src.get_integrity())
-			T.table_type = src.type
-			user.visible_message(span_danger("[user] flips over the [src]!"), span_notice("You flip over the [src]!"))
-			playsound(src, 'sound/items/trayhit2.ogg', 100)
-			qdel(src)
+	if(!can_flip)
+		return
+	user.visible_message(span_danger("[user] starts flipping [src]!"), span_notice("You start flipping over the [src]!"))
+	if(!do_after(user, max_integrity * 0.25))
+		return
+
+	var/obj/structure/flippedtable/flipped_table = new flipped_table_type(src.loc)
+	flipped_table.name = "flipped [src.name]"
+	flipped_table.desc = "[src.desc] It is flipped!"
+	flipped_table.icon_state = src.base_icon_state
+	var/new_dir = get_dir(user, flipped_table)
+	flipped_table.dir = new_dir
+	if(new_dir == NORTH)
+		flipped_table.layer = BELOW_MOB_LAYER
+	flipped_table.max_integrity = src.max_integrity
+	flipped_table.update_integrity(src.get_integrity())
+	flipped_table.table_type = src.type
+	if(custom_materials)
+		flipped_table.table_materials = src.custom_materials
```

Comment:
Just worth a test to see if this fixes the icon issue, try instead calling `flipped_table.set_custom_materials(src.custom_materials)`? It should handle the icon update, as well as properly making sure any durability modifiers are passed

Reply:
I assumed the icon was baked in actually but that was maybe a bit hasty. I'll try this tomorrow

Reply:
The icons aren't greyscale nope

Reply:
sad